### PR TITLE
Updated default Rust version to 1.49.0 to fix `bazel sync`

### PR DESCRIPTION
**Pull request by UebelAndre**
Having the `aarch64-apple-darwin` target in the list of default toolchains breaks `bazel sync` with the following error:
```output
INFO: Repository rust_darwin_aarch64 instantiated at:
  /Users/user/Code/cargo-raze/WORKSPACE.bazel:19:18: in <toplevel>
  /private/var/tmp/_bazel_user/32618c3132899251ff30687e197d6243/external/io_bazel_rules_rust/rust/repositories.bzl:110:24: in rust_repositories
  /private/var/tmp/_bazel_user/32618c3132899251ff30687e197d6243/external/io_bazel_rules_rust/rust/repositories.bzl:714:30: in rust_repository_set
Repository rule rust_toolchain_repository defined at:
  /private/var/tmp/_bazel_user/32618c3132899251ff30687e197d6243/external/io_bazel_rules_rust/rust/repositories.bzl:613:44: in <toplevel>
WARNING: Download from https://static.rust-lang.org/dist/rust-1.48.0-aarch64-apple-darwin.tar.gz failed: class com.google.devtools.build.lib.bazel.repository.downloader.UnrecoverableHttpException GET returned 404 Not Found
ERROR: An error occurred during the fetch of repository 'rust_darwin_aarch64':
   Traceback (most recent call last):
	File "/private/var/tmp/_bazel_user/32618c3132899251ff30687e197d6243/external/io_bazel_rules_rust/rust/repositories.bzl", line 578, column 44, in _rust_toolchain_repository_impl
		build_components = [_load_rust_compiler(ctx)]
	File "/private/var/tmp/_bazel_user/32618c3132899251ff30687e197d6243/external/io_bazel_rules_rust/rust/repositories.bzl", line 483, column 24, in _load_rust_compiler
		load_arbitrary_tool(
	File "/private/var/tmp/_bazel_user/32618c3132899251ff30687e197d6243/external/io_bazel_rules_rust/rust/repositories.bzl", line 442, column 17, in load_arbitrary_tool
		ctx.download(
Error in download: java.io.IOException: Error downloading [https://static.rust-lang.org/dist/rust-1.48.0-aarch64-apple-darwin.tar.gz] to /private/var/tmp/_bazel_user/32618c3132899251ff30687e197d6243/external/rust_darwin_aarch64/rust-1.48.0-aarch64-apple-darwin.tar.gz: GET returned 404 Not Found
```

With the release of Rust version `1.49.0` (and the tooling for `aarch64-apple-darwin`), `bazel sync` once again succeeds.
Closes https://github.com/bazelbuild/rules_rust/issues/506

Additionally, a Bazel target has been created for `//util:fetch_shas.sh` (`//util:fetch_shas`) and made compatible with MacOS which was used to update the sha256 map.

Users on MacOS will need to install [GNU parallel](https://www.gnu.org/software/parallel/) which can be done via
```bash
brew install parallel
```

### CODE REVIEW

**Comment by UebelAndre** [comment]
https://blog.rust-lang.org/inside-rust/2020/12/29/1.49.0-prerelease.html might actually change the nature of this pull request. If a compiler is released for `aarch64-apple-darwin` then instead the default version should change to `1.49` and the appropriate `sha256`s should be added. I still think this is worth while to merge since it fixes currently broken functionality.

**Comment by UebelAndre** [comment]
Ok, now that there *is* a compiler for `aarch64-apple-darwin`, this PR has changed to update the default version instead of removing the declaration of the toolchain.

**Comment by UebelAndre** [comment]
With `1.49.0` I ran into a new clippy error:
```
INFO: From Clippy test/clippy/ok_test_clippy.ok:
error: identical args used in this `assert_eq!` macro call
 --> test/clippy/src/lib.rs:9:20
  |
9 |         assert_eq!(2 + 2, 4);
  |                    ^^^^^^^^
  |
  = note: `#[deny(clippy::eq_op)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#eq_op

error: aborting due to previous error
```

A fix for this has also been added to this PR.

File: util/fetch_shas.sh [hunk 1]
```diff
@@ -2,17 +2,21 @@
...
-BETA_ISO_DATES="$(cat ./util/fetch_shas_BETA_ISO_DATES.txt)"
-NIGHTLY_ISO_DATES="$(cat ./util/fetch_shas_NIGHTLY_ISO_DATES.txt)"
-RUSTFMT_VERSIONS="$(cat ./util/fetch_shas_RUSTFMT_VERSIONS.txt)"
+# Detect workspace root
+if [[ -z "${BUILD_WORKSPACE_DIRECTORY}" ]]; then
+    SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+    BUILD_WORKSPACE_DIRECTORY="$( dirname "${SCRIPT_DIR}")"
+fi
+
+TOOLS="$(cat ${BUILD_WORKSPACE_DIRECTORY}/util/fetch_shas_TOOLS.txt)"
```
Comment (illicitonion) [inline util/fetch_shas.sh:14]: Now that you're using a variable in them, can you quote all of the args to cat in this block?

**Review by illicitonion — APPROVED** [review]
Thanks! One small nit and I think we're good to go

File: util/fetch_shas.sh (hunk 1 above)
Comment (UebelAndre) [reply util/fetch_shas.sh]: Updated

File: util/fetch_shas.sh [hunk 2]
```diff
@@ -46,16 +50,20 @@ enumerate_keys() {
...
+  echo ""
   echo "FILE_KEY_TO_SHA = {"
-  cat /tmp/reload_shas_shalist.txt | sort | awk '{print "    \"" $1 "\": \"" $2 "\","}'
+  cat /tmp/reload_shas_shalist.txt | sed '/^[[:space:]]*$/d' | sort | awk '{print "    \"" $1 "\": \"" $2 "\","}'
   echo "}"
+  rm /tmp/reload_shas_shalist.txt
 }
 
-echo "$(emit_bzl_file_contents $(enumerate_keys))" > ./rust/known_shas.bzl
+echo "$(emit_bzl_file_contents $(enumerate_keys))" > ${BUILD_WORKSPACE_DIRECTORY}/rust/known_shas.bzl
```
Comment (illicitonion) [inline util/fetch_shas.sh:69]:
```suggestion
echo "$(emit_bzl_file_contents $(enumerate_keys))" > "${BUILD_WORKSPACE_DIRECTORY}/rust/known_shas.bzl"
```

**Review by illicitonion — APPROVED** [review]

**Pull request merged**